🏃[e2e] add e2e target to Makefile and update book (#3068)
kubernetes-sigs/cluster-api: docs/book/src/developer/testing.md

is also useful if you simply want a local cluster to experiment with. After the tests run and succeed you are left
with a cluster in a known working state for experimentation.

* `FOCUS`: The `FOCUS` variable allows running of certain tests. By default, all tests are run if `FOCUS` is not provided. To run
a particular test, set focus to a regex string: e.g., `FOCUS='KCP upgrade'`.

### Running tests

`make test-e2e-images` will build the images for all providers that will be needed for the e2e test.

`make test-e2e` will run e2e tests by using whatever provider images already exist on disk.
After running `make test-e2e-images` at least once, this can be used for a faster test run if there are no provider code changes.

**Examples**

* `make test-e2e-images`

> * `make test-e2e FOCUS='upgrade' SKIP_RESOURCE_CLEANUP=true`

Reviewer: ```suggestion
* `make test-e2e GINKGO_FOCUS='upgrade' SKIP_RESOURCE_CLEANUP=true`
```

is also useful if you simply want a local cluster to experiment with. After the tests run and succeed you are left
with a cluster in a known working state for experimentation.

* `GINKGO_FOCUS`: The `GINKGO_FOCUS` variable allows running of certain tests. By default, all tests are run if `GINKGO_FOCUS` is not provided. To run
a particular test, set focus to a regex string: e.g., `GINKGO_FOCUS='KCP upgrade'`.

* `USE_EXISTING_CLUSTER`: The `USE_EXISTING_CLUSTER` variable allows using an existing management cluster with providers already installed.
For instance, after running e2e tests once with `SKIP_RESOURCE_CLEANUP=true`, the created management cluster can be used for the following e2e test runs by setting up `KUBECONFIG` variable.

### Running tests

> `make test-e2e-images` will build the images for all providers that will be needed for the e2e test.

Reviewer 2: ```suggestion
`make docker-buld-e2e` will build the images for all providers that will be needed for the e2e test.
```

`make test-e2e` will run e2e tests by using whatever provider images already exist on disk.

> After running `make test-e2e-images` at least once, this can be used for a faster test run if there are no provider code changes.

Reviewer 2: ```suggestion
After running `make docker-build-e2e` at least once, this can be used for a faster test run if there are no provider code changes.
```

**Examples**

> * `make test-e2e-images`

Reviewer 2: ```suggestion
* `make docker-build-e2e`
```
